applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1325,22 +1328,86 @@ int unicast_client_send(struct le_audio_encoded_audio enc_audio)
 		return -EINVAL;
 	}
 
-	if (ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-		ret = bt_iso_chan_get_tx_sync(&headsets[AUDIO_CH_L].sink_stream.ep->iso->chan,
-					      &tx_info);
-	} else if (ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-		ret = bt_iso_chan_get_tx_sync(&headsets[AUDIO_CH_R].sink_stream.ep->iso->chan,
-					      &tx_info);
-	} else {
+	if (!ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)
+		&& !ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
 		/* This can happen if a headset is reset and the state in
 		 * streamctrl hasn't had time to update yet
 		 */
 		LOG_DBG("No headset in stream state");
 		return -ECANCELED;
-	}
+	} else if (ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)
+			   && ep_state_check(headsets[AUDIO_CH_R].sink_stream.ep, BT_BAP_EP_STATE_STREAMING)
+			   && !channels_are_synched) {
+		/* We are transitioning from state where only one channel was streaming.
+		 * We know this because both channels are not synced. All channels needs to sync.
+		 */
+		
+		ret = iso_stream_send(enc_audio.data, data_size_pr_stream,
+						&headsets[AUDIO_CH_L], BT_ISO_TIMESTAMP_NONE);
+		if (ret) {
+			LOG_DBG("Failed to send data to left channel");
+		}
 
-	if (ret) {
-		LOG_DBG("Error getting ISO TX anchor point: %d", ret);
+		ret = bt_iso_chan_get_tx_sync(&headsets[AUDIO_CH_L].sink_stream.ep->iso->chan,
+					      &tx_info);
+		
+		if (ret) {
+			LOG_DBG("Error getting ISO TX anchor point: %d", ret);
+		}
+		LOG_WRN("Syncing channels, left sync %u", tx_info.ts);
+		
+		ret = iso_stream_send(enc_audio.data, data_size_pr_stream, &headsets[AUDIO_CH_R], tx_info.ts);
+
+		if (ret) {
+			LOG_DBG("Failed to send data to right channel");
+		}
+
+		channels_are_synched = true;
+
+	} else {
+
+		bool all_channels_in_stream_state = true;
+
+		if (ep_state_check(headsets[AUDIO_CH_L].sink_stream.ep, BT_BAP_EP_STATE_STREAMING))	{
+			
+			ret = iso_stream_send(enc_audio.data, data_size_pr_stream,
+						&headsets[AUDIO_CH_L], BT_ISO_TIMESTAMP_NONE);
```

Comment:
Can NONE be used here?

Reply:
Don't we need a ts here which is the same as we send for the other channel?

Reply:
If the channels were previously synced, the "time of arrival" based timestamp that the controller will assign these packets should be the same when they are sent at the "same" time to the controller.